Accelerate Answer Finding with LLM-Powered Website Analysis (#27)
groq/groq-api-cookbook: chat_with_website/README.md

# Website Guru

> Website Guru is a Streamlit application that allows you to interact with and ask questions about a given website using advanced language models. The app uses a combination of document loading, text splitting, vector storage, and retrieval chains to generate accurate responses based on the content of the specified website.

Reviewer: Suggest replacing "advanced language models" with Large Language Models (LLMs).

> Website Guru is a Streamlit application that allows you to interact with and ask questions about a given website using advanced language models. The app uses a combination of document loading, text splitting, vector storage, and retrieval chains to generate accurate responses based on the content of the specified website.

Reviewer: ```suggestion
Website Guru is a Streamlit application that allows you to interact with and ask questions about a given website using Large Language Models (LLMs). The app uses a combination of document loading, text splitting, vector storage, and retrieval chains to generate accurate responses based on the content of the specified website.

```

## Features

- Load and process web pages to extract information.
- Utilize advanced language models to answer questions about the content.
- Provide a user-friendly interface with Streamlit.

## Requirements

- Python 3.7 or higher
- Streamlit
- langchain
- langchain_groq
- langchain_community
- sentence-transformers
- faiss-cpu
- python-dotenv

## Installation

1. Install the required Python packages:
	
	pip install -r requirements.txt

2. Create a .env file in the project directory and add your Groq API key:

	GROQ_API_KEY=your_groq_api_key

## Usage

1. Run the Streamlit app:

	streamlit run app.py

2. Open your web browser and go to http://localhost:8501.

> 3. Enter the website address you want to analyze and choose a language model.

Reviewer: ```suggestion
3. Enter the website address you want to analyze and select LLM of your choice.

```
